src/rules.c

```diff
@@ -1,46 +1,185 @@
 #include "rules.h"
 #include "aggregate/expr/expression.h"
+#include "spec.h"
 
-IndexSpec **SchemaRules_g;
+arrayof(SchemaRule*) SchemaRules_g;
+TrieMap *ScemaPrefixes_g;
 
-SchemaRule *Rule_Create(ruleSettings *rulesopts, QueryError *status) {
-    RSExpr *e = ExprAST_Parse(rulesopts->expr, strlen(rulesopts->expr), status);
-    if (!e) {
-      return NULL;
-    }
+///////////////////////////////////////////////////////////////////////////////////////////////
 
-    SchemaRule *rule = rm_calloc(1, sizeof(*rule));
-    rule->expression = e;
-    rule->setting.expr = rm_strdup(rulesopts->expr);
+SchemaRule *SchemaRule_Create(SchemaRuleArgs *args, IndexSpec *spec, QueryError *status) {
+  SchemaRule *rule = rm_calloc(1, sizeof(*rule));
 
-    if (rulesopts->score) {
-      rule->setting.score = rm_strdup(rulesopts->score);
-    }
-    
-    if (rulesopts->lang) {
-      rule->setting.lang = rm_strdup(rulesopts->lang);
+  rule->type = rm_strdup(args->type);
```

Comment:
Use enum

Reply:
@ashtul it's not just the enum, when we support more follow types, we'll have to differentiate on how we perform it. Right now, the type is just a word - let's keep it like this for simplicity and do the extra abstraction when we're more stable.